app/models/run_test.rb

```diff
@@ -7,6 +7,18 @@ class RunTest < ActiveRecord::Base
   before_save :compile
   before_create :escape_jquery_html_characters
 
+  def screenshots(browser_type_id)
+    RunTestAction.where(run_test: self).map{|rta|
+      rta.action_statuses.where(browser_type_id: browser_type_id).map{|as| as.screenshot}.compact
+    }.compact #=> [['a1', 'b2', 'c3']]
+  end
+
+  def screenshot_count(browser_type_id)
+    imagearray = screenshots(browser_type_id)
+    imagearray[1].count #selects the inner array and counts across the images present
+  end
```

Comment:
Trying to think of how we can reduce database load. How do you feel about something like this? I think it's OK to call `.first` on the screenshot array, since there will never be more than one action_status per one browser_type's run_test_action. Then I'm hoping we can set screenshots 

```ruby
def screenshots(browser_type_id)
  self.run_test_actions.map { |rta|
    rta.action_statuses.where(browser_type_id: browser_type_id).map { |as| as.screenshot }.compact
  }.compact.first #=> ['a1', 'b2', 'c3']
end
```

Then we could do away with `screenshots_count` function and set `@screenshots = screenshots(bt)` in the Run controller in `def show`, and then call `@screenshots.count` -- that way the database is only called once